fedimint-core/src/encoding/mod.rs

```diff
@@ -228,10 +229,45 @@ where
         modules: &ModuleDecoderRegistry,
     ) -> Result<Self, DecodeError> {
         let len = u64::consensus_decode(d, modules)?;
-        (0..len).map(|_| T::consensus_decode(d, modules)).collect()
+
+        // `collect` under the hood uses `FromIter::from_iter`, which can potentially be
+        // backed by code like:
+        // <https://github.com/rust-lang/rust/blob/fe03b46ee4688a99d7155b4f9dcd875b6903952d/library/alloc/src/vec/spec_from_iter_nested.rs#L31>
+        // This can take `size_hint` from input iterator and pre-allocate memory
+        // upfront with `Vec::with_capacity`. Because of that untrusted `len`
+        // should not be used directly.
+        let cap_len = cmp::min(8_000, len);
```

Comment:
limit should also depend on `mem::size_of::<T>()`

Reply:
Right.